File: 11_sorts/02_insertion_sort/insertion_sort.py
arr  = [3, 5, 2, 4, 6, 1]

def insertion_sort(arr):
	for idx in range(1, len(arr)):
		j = idx

		while j > 0 and arr[j] < arr[j - 1]:
			arr[j], arr[j - 1] = arr[j - 1], arr[j]

			j -= 1

insertion_sort(arr)
print(arr)

File: 11_sorts/02_insertion_sort/insertion_sort.js
var arr  = [3, 5, 2, 4, 6, 1];

function insertion_sort(arr) {
	for (var i = 1; i < arr.length; i++) {
		var element = arr[i];
		var j = i;
		while (j > 0 && arr[j - 1] > element) {
			arr[j] = arr[j - 1];
			j--;
		}
		arr[j] = element;
	}
}

insertion_sort(arr);
console.log(arr);
